python/examples/agents/bee.py

```
@@ -83,62 +80,57 @@ def create_agent() -> BeeAgent:
-                "🛠️ System: ",
-                f"The code interpreter tool is enabled. Please ensure that it is running on {code_interpreter_url}",
-            )
-
-        reader.write("🛠️ System: ", "Agent initialized with LangChain Wikipedia tool.")
-
-        # Main interaction loop with user input
-        for prompt in reader:
-            # Run agent with the prompt
-            response = await agent.run(
-                BeeRunInput(prompt=prompt),
-                {
-                    "execution": {
-                        "max_retries_per_step": 3,
-                        "total_max_retries": 10,
-                        "max_iterations": 20,
-                    }
-                },
-            ).observe(observer)
-
-            reader.write("Agent 🤖 : ", response.result.text)
-
-    except Exception as e:
-        logger.error(f"Application error: {e!s}")
+    # Create agent
+    agent = create_agent()
+
+    # Log code interpreter status if configured
+    code_interpreter_url = os.getenv("CODE_INTERPRETER_URL")
+    if code_interpreter_url:
+        reader.write(
+            "🛠️ System: ",
+            f"The code interpreter tool is enabled. Please ensure that it is running on {code_interpreter_url}",
+        )
+
+    reader.write("🛠️ System: ", "Agent initialized with LangChain Wikipedia tool.")
+
+    # Main interaction loop with user input
+    for prompt in reader:
+        # Run agent with the prompt
+        response = await agent.run(
+            BeeRunInput(prompt=prompt),
+            BeeRunOptions(
+                execution=BeeAgentExecutionConfig(max_retries_per_step=3, total_max_retries=10, max_iterations=20)
+            ),
+        ).observe(observer)
+
+        reader.write("Agent 🤖 : ", response.result.text)
 
 
 if __name__ == "__main__":
-    # Run the async main function
-    asyncio.run(main())
+    try:
+        asyncio.run(main())
+    except FrameworkError as e:
+        print(e.explain())
+        sys.exit(1)
```

As mentioned in my other comment, this update hides the error messaging when run by `test_example_execution` from #420, this is a comparison of two example with the same failing:
```
FAILED tests/examples/test_examples.py::test_example_execution[agents/bee.py] - SystemExit: 1
FAILED tests/examples/test_examples.py::test_example_execution[basic.py] - beeai_framework.backend.errors.ChatModelError: Stream response is empty.
```

With the context of #420 I'm unsure if this is the best way to handle this error